client/client.cpp

```diff
@@ -114,7 +114,7 @@ void Client::socketConnected()
 void Client::resetClientDevice()
 {
     if (m_clientDevice) {
-        m_clientDevice->deleteLater();
+        QMetaObject::invokeMethod(m_clientDevice, "deleteLater", Qt::QueuedConnection);
```

Comment:
deleteLater() is already queued through the event loop, are you sure this change is ncessary?